src/provider/xds_grpc_sotw_provider.go

```diff
@@ -76,10 +78,25 @@ func (p *XdsGrpcSotwProvider) initXdsClient() {
 			logger.Info("Stopping xDS client watch for rate limit configurations")
 			break
 		}
+
+		d := p.getJitteredExponentialBackOffDuration()
+		logger.Infof("Sleeping for %s using exponential backoff\n", d)
+		time.Sleep(d)
 		conn = p.initializeAndWatch()
 	}
 }
 
+func (p *XdsGrpcSotwProvider) getJitteredExponentialBackOffDuration() time.Duration {
+	b := &backoff.Backoff{
+		Min:    10 * time.Second,
+		Max:    10 * time.Minute,
+		Factor: 0.5,
+		Jitter: true,
+	}
+	logger.Infof("Retry attempt# %f", b.Attempt())
```

Comment:
You'll want to move `b` outside of this function, otherwise you are resetting the backoff each time through the loop and you would only ever see one attempt.

Reply:
Ah thanks for pointing this Mike. You're right.